lib/src/list_extensions.dart

```diff
@@ -252,6 +252,10 @@ extension ListExtensions<E> on List<E> {
     }
     return true;
   }
+  
+  /// Returns an element at the given [index]
+  /// or `null` if the [index] is out of bounds of this list.
+  E? getOrNull(int index) => (index >= 0 && index < length) ? this[index] : null;
```

Comment:
Not sure I think this functionality is worth it. Too specialized. No clear use-case.

It allows you to do `list.getOrNull(n) ?? (throw ArgumentError("bad index")` instead of checking the index, but only if the list's element type is not null. It also easily hides errors in an index computation.

Reply:
`??` to hide errors in an index computation is easier than `index < list.length ? list[index] : value` but I think `list.getOrNull(index) ?? value` is also clear that index might out of bounds.
One problem is that it may hide errors if used to determine whether the element is null. In that case, we should use `list[index] == null` instead of `getOrNull` unless we need to deal with non-existent or null